app.js

```diff
@@ -9,85 +9,134 @@ const weather = {};
 
-    fetch(api)
+    fetch(currentAPI)
         .then(function(response){
-            let data = response.json();
-            return data;
+            return response.json();
         })
         .then(function(data){
-            weather.temperature.value = Math.floor(((data.main.temp - KELVIN) * (9/5)) + 32); //Converts from Kelvin to Fahrenheit
-            weather.description = data.weather[0].description; //MAIN INSTEAD OF DESCRIPTION?
-            weather.iconId = data.weather[0].icon;
-            weather.city = data.name;
-            weather.country = data.sys.country;
+            weather.temperature.currentValue = Math.floor(((data.main.temp - KELVIN) * (9/5)) + 32); //Converts from Kelvin to Fahrenheit
+            weather.currentDescription = data.weather[0].description; //MAIN INSTEAD OF DESCRIPTION?
+            weather.currentIconId = data.weather[0].icon;
+            weather.currentCity = data.name;
+            weather.currentCountry = data.sys.country;
         })
         .then(function(){
-            displayWeather();
+            displayCurrentWeather();
         });
-}
+};
 
-// DISPLAY WEATHER TO UI
-function displayWeather(){
-    iconElement.innerHTML = `<img src="icons/${weather.iconId}.png"/>`;
-    temperatureElement.innerHTML = `${weather.temperature.value}°<span>F</span>`;
-    descriptionElement.innerHTML = weather.description;
-    locationElement.innerHTML = `${weather.city}, ${weather.country}`;
-}
+// GET 3 DAY FORECAST FROM API ///////////////////////////////////////////////////////////////////
+function dayGetWeather(latitude, longitude){
+    let dayAPI = `https://api.openweathermap.org/data/2.5/forecast?lat=${latitude}&lon=${longitude}&appid=${key}`;
 
-// WHEN THE USER CLICKS ON THE .TEMPERATURE ELEMENT
-temperatureElement.addEventListener("click", function(){
-    if(weather.temperature.value === undefined) return;
+    console.log(dayAPI); // just for being able to double check answers
+
+    fetch(dayAPI)
+        .then(function(response){
+            return response.json();
+        })
+        .then(function(data){ 
+            weather.day1IconId = data.list[7].weather[0].icon;
+            weather.day2IconId = data.list[15].weather[0].icon;
+            weather.day3IconId = data.list[23].weather[0].icon;
+            weather.temperature.day1Value = Math.floor(((data.list[7].main.temp - KELVIN) * (9/5)) + 32);
+            weather.temperature.day2Value = Math.floor(((data.list[15].main.temp - KELVIN) * (9/5)) + 32);
+            weather.temperature.day3Value = Math.floor(((data.list[23].main.temp - KELVIN) * (9/5)) + 32);
+            weather.day1Description = data.list[7].weather[0].description;
+            weather.day2Description = data.list[15].weather[0].description;
+            weather.day3Description = data.list[23].weather[0].description;
+        })
+        .then(function(){
+            displayDayWeather();
```

Comment:
Same as above no need to keep chaining then handlers.

Reply:
If you have a good reason to please shut me up lol.

Reply:
Looks like it breaks when I combine them into a single then handler for response and data, but I was able to combine the display function into data. I'm gonna look into whats breaking it after I figure out the changes to day element selectors

Reply:
Nevermind I see the issue now. The second callback is dependent on the data. I'm just used to writing async/await instead of thenables.